pkg/controller/install/openshift/openshift.go

```diff
@@ -86,6 +87,168 @@ func EnsureOpenshiftIngress(c client.Client, scheme *runtime.Scheme, instance *s
+	if _, ok := cm.Data["service-ca.crt"]; ok {
+		// Do nothing
+		return nil
+	} else {
+		cm.Annotations = make(map[string]string)
+		cm.Annotations["service.alpha.openshift.io/inject-cabundle"] = "true"
+		cm.Namespace = namespace
+		if err := c.Update(context.TODO(), cm); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// EnsureOpenshiftRegistry ensures registry for openshift
+func EnsureOpenshiftRegistry(c client.Client, scheme *runtime.Scheme, instance *servingv1alpha1.Install) error {
+	namespace := instance.GetNamespace()
+	if routeExists, err := kindExists(c, "route", "route.openshift.io/v1", namespace); err != nil {
+		return err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return nil
+	}
+
+	if err := updateDeploymentController(c, namespace); err != nil {
+		return err
+	}
+
+	if err := updateConfigMapForRegistry(c, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateDeploymentController(c client.Client, namespace string) error {
+	depName := "controller"
+	cmName := "config-service-ca"
+	volName := "service-ca"
+
+	// Check the existance of configmap before updating the deployment/controller
+	cm := &v1.ConfigMap{}
+	if err := c.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: namespace}, cm); err != nil {
+		return err
+	}
+
+	if _, ok := cm.Data["service-ca.crt"]; ok {
+		found := &appsv1.Deployment{}
+		if err := c.Get(context.TODO(), types.NamespacedName{Name: depName, Namespace: namespace}, found); err != nil {
+			return err
+		}
+
+		/* Deployment exist so update the deployment controller with volume, volumeMount and env.
+		oc -n $ns set volume deployment/controller --add --name=service-ca --configmap-name=$configmap_name --mount-path=$mount_path
+		oc -n $ns set env deployment/controller SSL_CERT_FILE=$mount_path/$cert_name */
+
+		v := v1.Volume{
+			Name: volName,
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: cmName,
+					},
+				},
+			},
+		}
+
+		found.Spec.Template.Spec.Volumes = append(found.Spec.Template.Spec.Volumes, v)
+
+		vMount := v1.VolumeMount{
+			Name:      "service-ca",
+			MountPath: "/var/run/secrets/kubernetes.io/servicecerts",
+		}
+
+		for j := range found.Spec.Template.Spec.Containers {
+			found.Spec.Template.Spec.Containers[j].VolumeMounts = append(found.Spec.Template.Spec.Containers[j].VolumeMounts, vMount)
+		}
+
+		envVar := &v1.EnvVar{
+			Name:  "SSL_CERT_FILE",
+			Value: "/var/run/secrets/kubernetes.io/servicecerts/service-ca.crt",
+		}
+
+		for i := range found.Spec.Template.Spec.Containers {
+			found.Spec.Template.Spec.Containers[i].Env = append(found.Spec.Template.Spec.Containers[i].Env, *envVar)
+		}
+
+		if err := c.Update(context.TODO(), found); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func updateConfigMapForRegistry(c client.Client, namespace string) error {
```

Comment:
This should be a `Transformer` like `ingress` or `egress`. Because this CM is already a part of the upstream manifest, we can just change its values _before_ we apply it. It should greatly simplify the code as well.

Reply:
Handled..Thank you.